lib/blacklight/request_builders.rb

```diff
@@ -80,27 +87,24 @@ def solr_search_params(user_params = params || {}, processor_chain = search_para
     #                            added to the query post processing
     def build_solr_query(user_params, processor_chain, extra_params=nil)
       Deprecation.warn(RequestBuilders, "build_solr_query is deprecated and will be removed in blacklight-6.0. Use SearchBuilder#query instead")
-      Blacklight::SearchBuilder.new(user_params, processor_chain, self).query(extra_params)
+      search_builder(processor_chain).with(user_params).query(extra_params)
     end
 
     ##
     # Retrieve the results for a list of document ids
     def solr_document_ids_params(ids = [])
-      solr_documents_by_field_values_params blacklight_config.document_model.unique_key, ids
+      Deprecation.silence(Blacklight::RequestBuilders) do
```

Comment:
in 6.0, we kill `solr_documents_by_field_values_params` and move the impl into this method without the "flexibility" it offers.